caseologue_python/caseologue.py

```diff
@@ -228,7 +228,7 @@ def test_bad_uri(self):
                         "bad_rui",
                         r["entity"],
                         f"'{r['label']}'",
-                        "has a bad URI (entity) (regex :^http://edamontology.org/(data|topic|operation|format)_[0-9]\{4\}$)",
+                        r"has a bad URI (entity) (regex :^http://edamontology.org/(data|topic|operation|format)_[0-9]\{4\}$)",
```

Comment:
Do you think it would cause a problem if it was just
```suggestion
                        r"has a bad URI (entity) (regex :^http://edamontology.org/(data|topic|operation|format)_[0-9]{4}$)",
```
?  (sorry for bugging, no time to try it myself in Python right now, need to prepare a short talk in <30min)

Reply:
Otherwise maybe
```suggestion
                        r"has a bad URI (entity) (regex :^http://edamontology.org/(data|topic|operation|format)_[0-9]{{4}}$)",
```
??

Reply:
Sorry for all the guesses, now I looked it up and tested. The correct string is
```suggestion
                        "has a bad URI (entity) (regex :^http://edamontology.org/(data|topic|operation|format)_[0-9]{4}$)",
```
There is no need to escape {}, therefore the warning. And there are no backslashes we'd need to retain (those would either need `"\\1"` or `r"\1"`), therefore the string had the same problem as before.

If you want to see a lot of black screen:
<img width="996" height="583" alt="image" src="https://github.com/user-attachments/assets/3e3f4408-989c-480b-ba4e-cc8fd52e5950" />